src/ui/coverage/coverageInfo.ts

```diff
@@ -0,0 +1,294 @@
+		globalConfig.setCoverage(coverageGlobalMap);
+
+		renderer.renderInterface(vscode.window.visibleTextEditors, coverageGlobalMap);
+	}
+}
+
+/**
+ * Run the --coverage command (with -Z line-coverage) to generate the coverage output, parse the output and return the result
+ *
+ * @param command - the kani command to run along with the harness name
+ * @param harnessName - name of the harness
+ * @returns - the result of executing the --coverage command and parsing the output
+ */
+async function runCoverageCommand(command: string, harnessName: string): Promise<any> {
+	// Get the full resolved path for the root directory of the crate
+	const directory = path.resolve(getRootDir());
+	const commmandSplit = command.split(' ');
+
+	// Get args for the command to be executed
+	const args = commmandSplit.slice(1);
+
+	const options = {
+		shell: false,
+		cwd: directory,
+	};
+
+	const globalConfig = GlobalConfig.getInstance();
+	const kaniBinaryPath = globalConfig.getFilePath();
+
+	vscode.window.showInformationMessage(`Generating coverage for ${harnessName}`);
+	return new Promise((resolve, _reject) => {
+		execFile(kaniBinaryPath, args, options, async (_error: any, stdout: any, _stderr: any) => {
+			if (stdout) {
+				const parseResult = await parseKaniCoverageOutput(stdout);
+				resolve({ statusCode: 0, result: parseResult });
+			}
+		});
+	});
+}
+
+/**
+ * Search for the path to the report printed in Kani's output, and detect if we are in a remote
+ * enviroment before returning the result.
+ *
+ * @param stdout - Kani's standard output after running the visualize command
+ * @returns - undefined (error) or a result that indicates if the extension is executed on a local
+ *  or remote environment. The result includes a `path` (if local) or a `command` (if remote).
+ */
+async function parseKaniCoverageOutput(stdout: string): Promise<any | undefined> {
+	const kaniOutput: string = stdout;
+	const kaniOutputArray: string[] = kaniOutput.split('Coverage Results:\n');
+
+	const coverageResults = kaniOutputArray.at(1);
+
+	if(coverageResults === undefined) {
+		return '';
+	}
+
+	const coverageResultsArray = coverageResults.split('\n')!;
+
+	// If the global setting for showing output is on, show the output
+	const config = vscode.workspace.getConfiguration('Kani');
+	const showOutputWindow = config.get('showOutputWindow');
+
+	const terminal = vscode.window.createOutputChannel('Coverage Report');
+	terminal.appendLine(coverageResults);
+
+	// Use the value to show or hide the output window
+	if (showOutputWindow) {
+		// Open channel but don't change focus
+		terminal.show();
+	}
+
+	const coverage = parseCoverageData(coverageResultsArray);
+
+	// No command found from Kani
+	return coverage;
+}
+
+// Parse Coverage entry objects and convert it into CoverageMap or a map<file_path, map<line_number, status>>
+// We store this map as the global cache since it allows easy sorting and retrieval by file name, needed by VS Code
```

Comment:
```suggestion
// Parse `CoverageEntry` objects and convert it into `CoverageMap` or a `map<file_path, map<line_number, status>>`.
// We store this map as the global cache since it allows easy sorting and retrieval by file name, needed by VS Code.
```